netdata/kernel-collector
# CO-RE basis

### Description
This PR is bringing the basis for we move forward with eBPF CO-RE. I am also already converting the eBPF program `syncfs` to test the whole structure.

This current PR is assuming we will deliver for users the headers inside `co-re/includes/` and after this we will compile the code when the kernel has support for BTF.

### Test

1. Clone branch
2. Run `./tools/check-kernel-core.sh` to test your kernel. If you do not receive any warning, your host has support for BTF.
3. Install `bpftool` on your host.
4. Execute commands: `cd co-re && make`
5. Finally run: `./tests/syncfs`. If you do not have any warning, everything was fine during the tests.
